src/gravity/gravity.cpp

```diff
@@ -31,22 +32,38 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   std::string sys = pin->GetOrAddString("artemis", "coordinates", "cartesian");
   Coordinates coords = geometry::CoordSelect(sys, ndim);
 
-  std::string grav_type = pin->GetString("gravity", "type");
+  const Real gm = pin->GetOrAddReal("gravity", "gm", -Big<Real>());
```

Comment:
How about nan? via `Null<Real>()`.

Reply:
is parthenon fine with nan in the input file?

Reply:
My guess is you can't pass `NAN` in through the input file, but it should be fine as a default argument passed to `GetOrAddReal`

Reply:
If that's true, then this won't work across restarts?

Reply:
I'm not immediately sure how to do this except by adding a function like `GetOrAddOptionalReal` that returns an `std::optional<Real>` that defaults to being unitialized. That might be a lot of typing though, and I think `Null()` already gets the point across pretty well.

I think NaN does actually work across restarts because we've stored the `Real`-valued parameters in the restart file? I think what doesn't work is typing `nan` into the input parameter file and hoping that whatever machinery we use to reinterpret strings as `Real`s converts it into `NaN` in memory.